### ISSUE TITLE
Dualsense Battery Level is always showing 0%

### ISSUE THREAD

**Issue by ajmi982** [Ryochan7/DS4Windows#2777]
**Describe the bug**
Connect Dualsense. See 0% battery at all times

**To Reproduce**
See above

**Expected behavior**
Battery should show 100% when fully charged and decrease over time

**Screenshots and Logs**
If applicable, add screenshots to help explain your problem. Also, please add the most
log file (ds4windows_log.txt) from the Logs folder in your DS4Windows config folder.

**Desktop (please complete the following information):**
Dualsense
Windows 11
DS4Windows 3.2.4

**Additional context**
I believe the battery percent broke in the last 1-2 weeks. A number of updates have been coming recently so I can't pinpoint it exactly

**Comment by sonny-dev** [comment]
Same, been like this for a while

**Comment by mohamedaleya** [comment]
Same issue..

**Comment by Kanuan** [comment]
I've just tested, both wired and wireless, and DualSense battery measure is working as expected on the latest version.

Are your DualSense controllers updated to the latest firmware?

Official Sony Updater: https://controller.dl.playstation.net/controller/lang/en/fwupdater.html

**Comment by sabattle** [comment]
Same problem, stays at 0% whether connected via bluetooth or wired. Running latest DS4Windows and controller is updated to the latest firmware

**Comment by renanrfranca** [comment]
Same problem, can confirm am on the latest firmware, 0307. It's the v2 dualsense btw.

**Comment by costispavlou** [comment]
same issue as well here. both plugged and BT. 
![image](https://user-images.githubusercontent.com/57916603/226104695-c2a1ce30-5b5a-4016-a5f0-6ab1868c59f9.png)

**Comment by Splashier** [comment]
I also had the same issue. I wanted to update the firmware through the Dualsense firmware updater app, but it indicated the controller battery is too low for a firmware update. How I got around the issue is, I had plugged my PS5 controller into a wall mount charger (iPhone fast charger), even though the controller does not support the fast charging it charges anyways. I left it on charge for about 15 - 30 min maybe and when I plugged the controller back into the PC I was able to do a firmware update. After the update I opened DS4 and it indicated the battery percentage was at 15%.

**Comment by DistractionRectangle** [comment]
tl;dr If you use it via usb and always leave it plugged in, it presently will not charge, and eventually will discharge to 0%, and then stay there. 

On Version 3.0.18 (using the config edit suggested in #2471 to run it with the 6.x dotnet runtime), the dualsense charges when plugged in and does not enter an error state when disconnected (but battery still drains/discharges, and this is inline with default Windows behavior without hidhide/ds4windows running).

I haven't bisected the release window, but it seems an issue with how the controller is initialized/released was introduced after 3.0.18

<details><summary>Long Version</summary>
<h2>System/Setup Overview</h2>

- Windows 11 Pro Ver. 10.0.2261 Build 2261
- Latest Ds4Windows (x64) running in portable mode: 3.2.10
- Using dotnet runtime linked in release notes
- Setup hidhide, vigem, fakeinput using built in setup dialog
- Configured for exclusive access, and set to temporarily disable for specific applications which have native controller support//need access to raw controller (i.e. Sony's Firmware Updater)
- Initially using an outdated Dualsense firmware

<h2>My experience with this bug</h2>

Battery level consistently showing 0% on USB.
I figure this is perhaps a display bug, and passthrough to sony's firmware updater, which also confirmed I neded to charge it. So I used the computer for while and expect the dualsense to charge (ds4windows showed it as `0%+` which supposedly means it's charging). Charging does not happen. I figure it's maybe because of the lightbar, so I disable the lightbar in my default profile. Still does not charge.

At this point I figure it's a firmware bug, so I throw it on a wall charger for a bit and then update my firmware (I failed to notice if ds4windows registered a greater than zero battery percentage until after updating the firmware, so I don't know if there is a display/reporting bug for older firmwares).

After updating the firmware, I have a greater than zero battery percentage, success! But still it doesn't seem to charge when idle and without the lightbar. Further, when disconnected (either when stopping ds4windows or putting the pc to sleep), the lightbar becomes white and the dualsense actually discharges!

Some googling suggests that the white lightbar means the controller/firmware is in an error state. I disabled hidhide and turn off ds4windows to observe default behavior. Default windows behavior, the controller charges when idle and the pc is on, and when put to sleep, lightbar goes dark. Dualsense still does discharge when pc is asleep.

Knowing the error lies with ds4windows, I tried an older version (3.0.18), and charging now works! It still discharges on sleep, however it no longer enters an error state (lightbar goes dark instead of white, so it's behavior is inline with default windows behavior (and thus a separate behavior/problem).

</details>

**Comment by DistractionRectangle** [comment]
In addition to updating the firmware, I was pulling my hair out trying to diagnose a discharging issue (which would lead to a permanent 0% state - which brought me here in the first place) and discovered that resetting the dualsense via the reset pinhole button on the back resolved the problem.

**Comment by Noondy** [comment]
I read, that Steam is the problem. As long as Steam is running and the Dualsense is connected, it will decharge, even with USB connection.
I tested it with 2 controllers and Steam seems to be the problem. When Steam isn't running, the controllers are charged while connected via USB. With Steam on, the battery level is always sinking.
Maybe we should all go and ask Steam the big question, why the controllers are drained while using Steam? I dont think, that always 0% and decharging while playing is good for the battery life...

**Comment by Heywhat50** [comment]
I havev the same Issue, everything is updated, I don't use steam, I reseted it from the pinhole, nothing worked for me. Here is the log file if someone wants to check it

[DS4.txt](https://github.com/Ryochan7/DS4Windows/files/13457032/DS4.txt)

**Comment by Ryochan7** [comment]
Not much of a point leaving this open. All this time has passed and there has been no help on the issue. The source code is open for a reason; and it is not just so others can pillage and profit off my autism. Why bother acting like there is some community here?

It might a hardware issue anyway. It obviously works for me so there is nothing for me to do.

![hey_peter](https://github.com/Ryochan7/DS4Windows/assets/581968/7ebe6f0d-ec77-47f6-859f-dc0e2e10dec4)
